core/src/main/java/org/infinispan/persistence/manager/PersistenceManagerImpl.java

```diff
@@ -585,7 +621,12 @@ private void initializeWriter(StoreConfiguration cfg, CacheWriter writer, Initia
       if (writer != null) {
          if (writer instanceof DelegatingCacheWriter)
```

Comment:
@wburns Good point. 

This has made me think about the compatibility of a transactional mode and a singleton cache writer in general. As it is the Tx coordinator who performs the store writers during the prepare phase of the Tx, the cluster coordinator (singleton) is not guaranteed to be involved in a Tx in distributed mode. Therefore, I think we will have to have some kind of exception/log message if a user specifies a store store to be singleton and transactional in distributed mode.  

In local and replicated mode, there is no conflict (AFAIK) between a singleton and transactional store.